docs/onnx-conversion.md

```diff
@@ -1,75 +1,251 @@
-# ONNX Conversion for SPLADE++
+# End to End ONNX Conversion for SPLADE++ Ensemble Distil
 This MD file will describe steps to convert particular PyTorch models (i.e., [SPLADE++](https://doi.org/10.1145/3477495.3531857)) to ONNX models and options to further optimize compute graph for Transformer-based models. For more details on how does ONNX Conversion work and how to optimize the compute graph, please refer to [ONNX Tutorials](https://github.com/onnx/tutorials#services).
 
+All scripts are available for reference under in the following directory:
+```
+src/main/python/onnx
+```
+
+The following tutorial will be using the scripts available in the above directory directly.
+
 ## Setups
-- pip requirements:
-    ```
-    onnx                          1.13.1
-    onnxoptimizer                 0.3.10
-    onnxruntime                   1.11.1
-    ```
-- SPLADE [Git repo](https://github.com/naver/splade):
-   ```bash
-   git clone https://github.com/naver/splade.git
-   ```
+
+pip requirements:
+```bash
+pip install torch transformers onnx onnxruntime onnxoptimizer
+```
 
 ## Converting from PyTorch models to ONNX model
 The following sections will describe how to convert SPLADE++ model to ONNX model. The steps are as follows:  
 
-### Load the corresponding HuggingFace model
+### Run the End to End PyTorch to ONNX Conversion
+Loading and running is done easily with argparse in the following script:
+```
+src/main/python/onnx/convert_hf_model_to_onnx.py
+```
+
+All that needs to be provided is the model_name as seen on huggingface. In our example we will be loading the SPLADE++ Ensemble Distil model found here:
+```
+naver/splade-cocondenser-ensembledistil
+```
+
+To run the script and produce the onnx model, run the following sequence of commands:
+```bash
+# Begin by going to the appropriate directory
+cd src/main/python/onnx
+# Now run the script
+python3 convert_hf_model_to_onnx.py --model_name naver/splade-cocondenser-ensembledistil
+```
+
+So what actually happens under the hood? The following sections will discuss the key parts of the above script:
+
+### Getting Output Specificaton from the Model
+
+Since we want our script to be generic, and be able to load any huggingface model with simplicity, we make it possible to fetch all required information from the model output structure with the following code:
+
 ```python
-from splade.models.transformer_rep import Splade
-import torch
+with torch.no_grad():
+    outputs = model(**test_input) # We provide the text test input and the model object.
```

Comment:
Where does model come from here? also test output?

Reply:
provided in the function definition. I guess I can just copy the full function definitions to make it easier.